### ISSUE TITLE
bug(windows/developer): Keyman for Windows crashes if .kxx file found during upgrade

### ISSUE THREAD

**Issue by MakaraSok** [keymanapp/keyman#4897]
**Describe the bug**

On a user machine,  a fresh installation of Keyman for Windows 14 given an error during the installation. Here is the crash message:
![image](https://user-images.githubusercontent.com/28331388/115178743-1e91b380-a0fc-11eb-8754-93578b0ed14c.png)

Details for error report for Keyman app:
Application Identifier: kmshell-14.0.271
Error Identifier: 3C3DD2865B504AFACC437C2EACE9517D

After closing the box, Keyman can be openned and it works just fine.

Now when the user tried and installed Keyman Developer, everything went through to the end and the installation completed successfully. It crashes though when opened.

![image](https://user-images.githubusercontent.com/28331388/115178943-77614c00-a0fc-11eb-8939-baf0cb589dc4.png)

Details for error report for Keyman app:
Application Identifier: tike-14.0.271
Error Identifier: ADE9ACB7D7834997520344810BAFDFC5

---

**Keyman for Windows:**
 - OS: Windows 10 Pro 20H2 x64, OS build 19042.928
 - Keyman version: 14.0.271
 - Target Application: Keyman

---

**Additional context**

This error happens very consistently for the user rendering him not able to do any keyboard development.

diag report for @mcdurdin https://drive.google.com/file/d/1kVE7IYWVcwX4ntfjkNBeSzRmU9rd2t6H/view?usp=sharing

**Comment by mcdurdin** [comment]
The first two [Keyman issues](https://sentry.keyman.com/share/issue/b7ae085456414f8f95d3c8f4abdd45c5/) look like a conflict between GeezWord 13 and Keyman 14. You may need to uninstall GW13 before continuing.

The [Developer issue](https://sentry.keyman.com/share/issue/c90859309c8a4a5bbe11cbf517367466/) is unrelated. It may be possible to resolve this by deleting the last open Keyman Developer Project .kpj.user file which it appears been corrupted (I don't know which file that would be from the crash report).

It might be helpful to capture a copy of the .kpj.user file if we can figure out how it got corrupted?

**Comment by mcdurdin** [comment]
From the diag report, the file that is most likely causing the problem is 

`C:\Users\Yemane\Documents\KEYMAN DEVELOPER\Projects\geezword tigrinya\geezword tigrinya mobile - 04-09-2021-FinalGitHub -Color\geezword_tigrinya.kpj.user`

**Comment by mcdurdin** [comment]
Community forum post (please always link these in the GH report @MakaraSok 😁): https://community.software.sil.org/t/installation-of-keyman-developer-14-crushed/4687

**Comment by mcdurdin** [comment]
This issue arises because the .kpj.user file has become corrupted at the end of the file. I examined the file in question and it ended like this:

```xml
</KeymanDeveloperProjectUser>
er>
```

I found one other report online of similar corruption: https://www.daniweb.com/programming/software-development/threads/474839/writing-to-xml-file-causes-corrupt-file

But I have not been able to reproduce this here.

One simple workaround would be to delete the xml file before writing to it.
